### PR TITLE
Azure OpenAI API 

### PR DESCRIPTION
**Pull request by shubham184**
## Summary

Fixes

Depends on

## Test plan

-

## Checklist

- [x] My code follows the code style of this project and passes {`style_guide_run_command`}.
- [ ] My change requires a change to the documentation.
- [ ] I have [updated the documentation](https://docs.superagent.sh/) accordingly.
- [ ] My change has adequate [Unit Test coverage]({TestCoverage}).

### CODE REVIEW

File: app/lib/agents.py [hunk 1]
```diff
@@ -131,6 +140,34 @@ def _get_llm(self) -> Any:
...
+                            on_llm_end_=self.on_llm_end,
+                            on_chain_end_=self.on_chain_end,
+                        )
+                    ],
+                )
+                if self.has_streaming
+                else AzureChatOpenAI(
+                    deployment_name=self.llm["model"],
+                    openai_api_key=self._get_api_key(),
+                    #openai_api_key="71d7dba9e0254d048eda1f8d5ef369da",
```
Comment (homanp) [inline app/lib/agents.py:164]: Remove this please!

File: app/lib/agents.py [hunk 2]
```diff
@@ -131,6 +140,34 @@ def _get_llm(self) -> Any:
                 else Cohere(cohere_api_key=self._get_api_key(), model=self.llm["model"])
             )
 
+        if self.llm["provider"] == "azure-openai":
+            return (
+                AzureChatOpenAI(
+                    openai_api_key=self._get_api_key(),
+                    openai_api_base="https://fujitsuesgopenai.openai.azure.com/",
```
Comment (homanp) [inline app/lib/agents.py:147]: Please use environment variables for this.

File: app/lib/agents.py [hunk 3]
```diff
@@ -131,6 +140,34 @@ def _get_llm(self) -> Any:
                 else Cohere(cohere_api_key=self._get_api_key(), model=self.llm["model"])
             )
 
+        if self.llm["provider"] == "azure-openai":
+            return (
+                AzureChatOpenAI(
+                    openai_api_key=self._get_api_key(),
+                    openai_api_base="https://fujitsuesgopenai.openai.azure.com/", 
+                    openai_api_type="azure",
```
Comment (homanp) [inline app/lib/agents.py:148]: Please use environment variables for this.

File: app/lib/agents.py [hunk 4]
```diff
@@ -131,6 +140,34 @@ def _get_llm(self) -> Any:
...
+                    openai_api_version="2023-03-15-preview",
```
Comment (homanp) [inline app/lib/agents.py:149]: Please use environment variables for this.

File: app/lib/agents.py [hunk 5]
```diff
@@ -18,6 +18,8 @@
 from app.lib.prisma import prisma
 from app.lib.prompts import default_chat_prompt
 
+import logging
```
Comment (homanp) [inline app/lib/agents.py:21]: Does your PR need logging enabled? If not, please remove.

**Review by homanp — CHANGES_REQUESTED** [review]
@shubham184 Big thanks for this PR. I've added some comments to it. Please have a look and afterwards I can merge this in.

**Comment by homanp** [comment]
@shubham184 thanks for this contribution, I left some comments for you. As soon as these are resolved I'll run some tests and merge.

File: app/lib/agents.py (hunk 1 above)
Comment (shubham184) [reply app/lib/agents.py]: done

File: app/lib/agents.py (hunk 2 above)
Comment (shubham184) [reply app/lib/agents.py]: done

File: app/lib/agents.py (hunk 3 above)
Comment (shubham184) [reply app/lib/agents.py]: done

File: app/lib/agents.py (hunk 4 above)
Comment (shubham184) [reply app/lib/agents.py]: done

**Review by homanp — APPROVED** [review]
Thanks for making the changes @shubham184 !

File: app/lib/agents.py (hunk 5 above)
Comment (shubham184) [reply app/lib/agents.py]: removed

**Comment by homanp** [comment]
@shubham184 could you run `make lint` and `make format`. I have resolved the merge conflicts. After that I can will merge this PR.

**Comment by homanp** [comment]
@shubham184 did some formatting changes and merge this! Thank you for your contribution!

**Pull request merged**